examples/conform.cpp

```diff
@@ -0,0 +1,164 @@
+//
+// You may obtain a copy of the Apache License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the Apache License with the above modification is
+// distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the Apache License for the specific
+// language governing permissions and limitations under the Apache License.
+//
+
+// Example OTIO script that reads a timeline and then relinks clips
+// to movie files found in a given folder, based on matching names.
+//
+// Demo:
+//
+// % ls -1R
+// editorial_cut.otio
+// media/
+//    shot1.mov
+//    shot17.mov
+//    shot99.mov
+//
+// % conform editorial_cut.otio media conformed.otio
+// Relinked 3 clips to new media.
+// Saved conformed.otio with 100 clips.
+//
+// % diff editorial_cut.otio conformed.otio
+// ...
+
+#include "util.h"
+
+#include <opentimelineio/clip.h>
+#include <opentimelineio/externalReference.h>
+#include <opentimelineio/timeline.h>
+
+#include <iostream>
+
+namespace otio = opentimelineio::OPENTIMELINEIO_VERSION;
+
+// Look for media with this name in this folder.
+std::string find_matching_media(std::string const& name, std::string const& folder)
+{
+    // In this case we're looking in the filesystem.
+    // In your case, you might want to look in your asset management system
+    // and you might want to use studio-specific metadata in the clip instead
+    // of just the clip name.
+    // Something like this:
+    // shot = asset_database->find_shot(
+    //    otio::any_cast<std::map<std::string, std::string> >(clip->metadata()["mystudio"])["shotID"]);
+    // new_media = shot->latest_render("mov");
+    
+    const auto matches = glob(folder, name + ".*");
+    
+    if (matches.size() == 0)
+    {
+        //std::cout << "DEBUG: No match for clip '" << name << "'" << std::endl;
+        return std::string();
+    }
+    if (matches.size() == 1)
+    {
+        return matches[0];
+    }
+    else
+    {
+        std::cout << "WARNING: " << matches.size() << " matches found for clip '" <<
+            name << "', using '" << matches[0] << "'";
+        return matches[0];
+    }
+}
+
+// Look for replacement media for each clip in the given timeline.
+//
+// The clips are relinked in place if media with a matching name is found.
+//
+// Note the use of otio::SerializableObject::Retainer to wrap the timeline,
+// it provides a safe way to manage the memory of otio objects by keeping an
+// internal reference count. For more details on the usage of Retainers see
+// the C++ documentation:
+// https://opentimelineio.readthedocs.io/en/latest/cxx/cxx.html
+int conform_timeline(
+    otio::SerializableObject::Retainer<otio::Timeline> const& timeline,
+    std::string const& folder)
+{
+    int count = 0;
+    
+    otio::ErrorStatus error_status;
+    const auto clips = timeline->clip_if(&error_status);
```

Comment:
This is more a comment for the other PR, but you were proposing adding `child_if` which stops at the first matching child -- I wonder if this should be `clips_if` rather than the singular form to make it clear that it won't stop at the first thing it finds but rather goes and builds an entire list?  What do you think?  @darbyjohnston @meshula

Reply:
I was thinking the same thing, should I do that as part of the other PR or a new one? It looks like there are about 30 occurences of ```clip_if``` in the otio code.

Reply:
new PR I'd say!